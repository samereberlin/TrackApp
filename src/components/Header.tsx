import React from 'react';
import {
  Pressable,
  StyleSheet,
  Text,
  TextStyle,
  View,
  ViewStyle,
} from 'react-native';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'stretch',
    backgroundColor: '#f8fafa',
    borderBottomColor: '#f1f1f1',
    borderBottomWidth: 2,
  } as ViewStyle,
  label: {flex: 1, color: '#333333', fontSize: 20, padding: 16} as TextStyle,
  backButton: {justifyContent: 'center', paddingLeft: 16, paddingRight: 16},
  backButtonIcon: {color: '#aabbbf', fontSize: 40},
});

export interface HeaderProps {
  label: string;
  buttonCallback?: Function;
}

const Header: React.FC<HeaderProps> = ({label, buttonCallback}) => (
  <View style={styles.container}>
    <Text style={styles.label}>{label}</Text>
    {buttonCallback && (
      <Pressable style={styles.backButton} onPress={() => buttonCallback()}>
        <Text style={styles.backButtonIcon}>Ã—</Text>
      </Pressable>
    )}
  </View>
);

export default Header;
