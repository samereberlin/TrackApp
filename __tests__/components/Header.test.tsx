import React from 'react';
import {create} from 'react-test-renderer';
import {fireEvent, render} from '@testing-library/react-native';

import Header from '../../src/components/Header';

describe('Header', () => {
  it('matches the snapshot', () => {
    const screen = create(<Header label="foo" />).toJSON();
    expect(screen).toMatchSnapshot();
  });
  it('renders correctly', () => {
    const screen = render(<Header label="foo" buttonCallback={() => {}} />);
    expect(screen.queryByText('foo')).toBeTruthy();
    expect(screen.queryByText('×')).toBeTruthy();
  });
  it('runs the button callback on click', () => {
    const mockCallback = jest.fn();
    const screen = render(<Header label="foo" buttonCallback={mockCallback} />);
    expect(mockCallback).not.toHaveBeenCalled();
    fireEvent.press(screen.getByText('×'));
    expect(mockCallback).toHaveBeenCalled();
  });
});
