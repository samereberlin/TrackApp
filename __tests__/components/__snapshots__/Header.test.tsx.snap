// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#f8fafa",
      "borderBottomColor": "#f1f1f1",
      "borderBottomWidth": 2,
      "flexDirection": "row",
    }
  }
>
  <Text
    numberOfLines={1}
    style={
      Object {
        "color": "#333333",
        "flex": 1,
        "fontSize": 20,
        "padding": 16,
      }
    }
  >
    foo
  </Text>
</View>
`;
